inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -365,8 +385,11 @@ def execute_from_manifest(self,
                 # The download-only mode only downloads the packages without installing them.
                 # Since there is no installation, there will be no changes in the package status or version.
                 # The apt history.log also doesn't record any changes. Therefore we can skip saving granular log.
-                elif self.sota_mode != 'download-only':
-                    self._update_logger.save_granular_log_file()
+                # Always save the granular log in TiberOS.
+                # TODO: Remove Mariner when confirmed that TiberOS is in use
+                elif self.sota_mode != 'download-only' or detect_os() == LinuxDistType.TiberOS.name \
```

Comment:
would be clearer to have another `elif` clause that saves the log if in TiberOS--was confusing to read at first, seemed like the download-only was related to the TiberOS.  also explain more in comment why we save log in TiberOS with download-only but not other OSes

Reply:
The granular log in TiberOS differs from that in other OSes. 
- In other OSes, the granular log is used to record the package level information.
- In TiberOS, it is used to record detail of SOTA such as detail.Status, failure reason and OS version.

The reason for not saving the granular log in other OSes, such as Ubuntu, is because the download-only mode in other OSes only downloads the packages. The granular log is used to record the package level information. Since download-only mode only downloads the package and doesn't install it, there is no installed package level information to be recorded. 

In TiberOS, however, the download-only mode is used to download the artifacts from the OCI registry. The granular log is enabled in TiberOS with the download-only mode because we want to record the artifact download failure.

Reply:
Will add this to comment as well.